nova_checker/src/lib.rs

```diff
@@ -0,0 +1,231 @@
+use codespan_reporting::diagnostic::{Diagnostic, Label};
+use hashbrown::{hash_set, HashMap, HashSet};
+use nova_parser::ast::{BinaryOp, Binding, BindingLevel, Expr, Stmt, UnaryOp};
+use std::{cell::RefCell, fmt::Display, hash::BuildHasher, rc::Rc};
+
+#[derive(Debug)]
+pub struct Env<'a> {
+    pub parent: Option<Rc<RefCell<Env<'a>>>>,
+    pub entries: HashMap<&'a str, Type>,
+}
+
+impl<'a> Env<'a> {
+    pub fn new() -> Self {
+        Self {
+            parent: None,
+            entries: HashMap::new(),
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct Checker<'a> {
+    pub source: &'a str,
+    pub diagnostics: Vec<Diagnostic<usize>>,
+}
+
+#[derive(Debug, Clone)]
+pub enum Error {
+    InvalidBinaryOperator {
+        expr: Expr,
+        lhs_t: Type,
+        rhs_t: Type,
```

Comment:
We can use fully written out names...?
It looks like this is C.
```suggestion
        expression: Expression,
        left_hand: Type,
        right_hand: Type,
```